src/common/misc.c

```diff
@@ -1071,3 +1074,28 @@ daos_prop_copy(daos_prop_t *prop_req, daos_prop_t *prop_reply)
 	}
 	return rc;
 }
+
+static __thread struct {
+	uint32_t	host_id;
+	uint32_t	thread_id;
+	uint64_t	hlc;
+} dtx_id_tls;
+
+void
+daos_dti_gen(struct dtx_id *dti, bool zero)
+{
+	if (zero) {
+		memset(dti, 0, sizeof(*dti));
+	} else {
+		/* uuid_generate() is too slow to generate the uuid for dti_uuid
+		 * now use {host_id, thread_id, hlc} to replace it.
+		 */
+		if (dtx_id_tls.hlc == 0) {
+			dtx_id_tls.host_id = gethostid();
```

Comment:
hmm, just tested that on different boro nodes, the gethostid() returns same value. it should be unique but not sure why same on boro nodes.